bin/rucio

```diff
@@ -327,36 +356,81 @@ def list_dataset_replicas(args):
     def _append_result(dsn, replica):
         if dsn not in result:
             result[dsn] = {}
-        result[dsn][replica['rse']] = [replica['rse'], replica['available_length'], replica['length']]
-
-    if len(args.dids) == 1:
-        scope, name = get_scope(args.dids[0], client)
-        dmeta = client.get_metadata(scope, name)
-        _fetch_datasets_for_meta(meta=dmeta)
+        if cli_config == 'rich':
+            result[dsn][replica['rse']] = [replica['rse'], str(replica['available_length']), str(replica['length'])]
+        else:
+            result[dsn][replica['rse']] = [replica['rse'], replica['available_length'], replica['length']]
+
+    if cli_config == 'rich':
+        status = Status('Fetching dataset replicas', spinner=CLITheme.SPINNER, spinner_style=CLITheme.SPINNER_STYLE, console=console)
```

Comment:
Is there a way to define a placeholder for status when rich isn't enabled? There's a ton of repeated code between the if/else and that's rife for inconsistency problems.

Reply:
Not sure about a placeholder but maybe switching to `status.start()` instead of the context manager might help with that. Then the if/else conditions would need to be in the core loops collecting the data since some of the logic between storing the rows for tabulate and rich are slightly different.

Reply:
I think that's a better solution!